packages/space-opera/src/components/poster_controls/poster_controls.ts

```diff
@@ -24,80 +24,99 @@ import {safeDownloadCallback} from '@google/model-viewer-editing-adapter/lib/uti
 
   @internalProperty() poster?: string;
+  @internalProperty() className: string = '';
+  @internalProperty() snackBody: string = '';
+  @internalProperty() cameraSnippet: Camera = {};
 
   stateChanged(state: State) {
     this.poster = getConfig(state).poster;
+    this.cameraSnippet = getCamera(state);
   }
 
   render() {
     return html`
-      <me-expandable-tab tabName="Poster">
-        <div slot="content">
-          <div class="ButtonContainer">
-            <mwc-button unelevated
-              @click="${this.onCreatePoster}">Create Poster</mwc-button>
-          </div>
-          ${
+<me-expandable-tab tabName="Poster">
+  <div slot="content">
+    <me-card title="Poster Creation">
+      <div slot="content">
+        <mwc-button unelevated class="PosterButton" 
+          @click="${
+        this.onCreatePoster}">Generate Poster At Initial Camera</mwc-button>
+        <div class="PosterHelperButtons">
+        ${
     !!this.poster ? html`
-            <div class="ButtonContainer">
-              <mwc-button unelevated
-                @click="${this.onDownloadPoster}">Download</mwc-button>
-            </div>
-            <div class="ButtonContainer">
-            <mwc-button unelevated
-              @click="${this.onDisplayPoster}">Display Poster</mwc-button>
-          </div>
-          <div class="ButtonContainer">
-            <mwc-button unelevated
-              @click="${this.onDeletePoster}">Delete Poster</mwc-button>
-          </div>` :
+        <mwc-button unelevated class="PosterButton"
+          @click="${this.onDownloadPoster}">Download Poster</mwc-button>
+        <mwc-button unelevated class="PosterButton"
+          @click="${this.onDisplayPoster}">Redisplay Poster</mwc-button>
+        <mwc-button unelevated class="PosterButton"
+          @click="${this.onDeletePoster}">Delete Poster</mwc-button>` :
                     html` `}
         </div>
-      </me-expandable-tab>
+      </div>
+    </me-card>
+  </div>
+</me-expandable-tab>
+<div class="${this.className}" id="snackbar">${this.snackBody}</div>
         `;
   }
 
   async onCreatePoster() {
     const modelViewer = getModelViewer()!;
     if (!modelViewer)
       return;
-    const posterUrl =
-        createSafeObjectURL(await modelViewer.toBlob({idealAspect: true}));
-    reduxStore.dispatch(dispatchSetPoster(posterUrl.unsafeUrl));
-    const currentOrbit = getCameraState(modelViewer).orbit;
-    const currentFieldOfViewDeg = getCameraState(modelViewer).fieldOfViewDeg;
-    reduxStore.dispatch(
-        dispatchSaveCameraOrbit(currentOrbit, currentFieldOfViewDeg));
+
+    // if we've already set the initial camera, use it
+    if (this.cameraSnippet.orbit !== undefined) {
+      const initialOrbit = this.cameraSnippet.orbit;
+      modelViewer.cameraOrbit = getOrbitString(initialOrbit);
+      modelViewer.jumpCameraToGoal();
+      modelViewer.requestUpdate();
+      requestAnimationFrame(async () => {
```

Comment:
is this the correct way to use requestAnimationFrame?

Reply:
LGTM, though I think you're missing a `;`. I think `requestUpdate()` is superfluous (since RAF invokes it), but you can test that pretty easily.

Reply:
done